src/orion/algo/random.py

```diff
@@ -62,4 +64,11 @@ def observe(self, points, results):
 
         A simple random sampler though does not take anything into account.
         """
-        pass
+        self.alreadyrun += len(points)
+
+    @property
+    def is_done(self):
```

Comment:
This is not specific to RandomSearch and would affect any algorithm. I would instead add this check in Experiment.is_done.

Reply:
I was thinking to such capacity check in Experment.is_done too, but consider that different algorithm may have different ways to decide whether the search space is already empty or not, like for Hyperband. So I decided to put this in the Random first and we can think if could define a generic way for all.

Reply:
Indeed some algo may have specific is_done procedures, but the one based on fully explored search space is generic and thus should be in Experiment in my opinion. Algorithms will either be done before search space is fully explored or in worst case when it gets fully explored. So in some sense algorithm.is_done have precedence over experiment.is_done.